ocrd_tool.md

```diff
@@ -19,6 +19,27 @@ Optionally, workflow processors can be notified that this file is potentially
 large and static (e.g. a fixed dataset or a precomputed model) and should be cached
 indefinitely after download by setting the `cacheable` property to `true`.
 
+The filename itself, i.e. the concrete value `<fpath>` of a file parameter
+should be resolved in the following way:
+
+* If `<fpath>` is an `http`/`https` URL: Download to a temporary directory (if
+  `cacheable==False`) or a semi-temporary cache directory (if `cacheable==True`)
+* If `<fpath>` is an absolute path: Use as-is.
+* If `<fpath>` is a relative path, try resolving the following paths and return
+  the first one found if any, raise a `FileNotFoundError` otherwise:
+  * `$CWD/<fpath>`
+  * If an environment variable is defined that has the name of the processor in
+    upper-case and with `-` replaced with `-` and followed by `_PATH` (e.g. for a processor
+    `ocrd-dummy`, the variable would need to be called `OCRD_DUMMY_PATH`):
+    * Split the variable value at `:` and try to resolve by prepending each token
+      with `<fname>`
+  * `$VIRTUAL_ENV/share/<name-of-processor>/<fpath>`
+  * `$HOME/.local/share/<name-of-processor>/<fpath>`
+  * `$HOME/.config/<name-of-processor>/<fpath>`
```

Comment:
Since this looks like [XDG Base Directory standard](https://unix.stackexchange.com/a/313001), maybe allow overriding these by `$XDG_DATA_HOME` and `$XDG_CONFIG_HOME`?

Reply:
True, though I've never seen anyone change either, ever and

`~/.config` and even
`$HOME/.config`
is shorter than
`$XDG_CONFIG_HOME`.

But you're right, I'm a big fan of XDG, should call them as such.

Reply:
But I did not mean _replace_ with the variable expansion, only _try_ it first. (IIUC, XDG requires to allow these as override for the fixed path name. They don't need to have any value. They don't have on Ubuntu.)

Reply:
we're in agreement. if the wording is unclear, I'm open to suggestions.

Reply:
thx